src/crons/websocket/room.key.generator.ts

```diff
@@ -1,23 +1,37 @@
 import { EventsCustomSubscribePayload } from 'src/endpoints/events/entities/events.custom.subscribe';
 import { TransactionCustomSubscribePayload } from 'src/endpoints/transactions/entities/dtos/transaction.custom.subscribe';
+import { TransferCustomSubscribePayload } from 'src/endpoints/websocket/entities/transfers.custom.payload';
 
 export class RoomKeyGenerator {
   public static generate(
     prefix: string,
     data: Record<string, any>,
     dtoClass: Function,
   ): string[] {
-    // get allowed keys from DTO (with caching)
     const allowedKeys = this.getKeys(dtoClass);
 
     // extract only key-value pairs that exist in the data and are non-null
     const activeFilters: { key: string; value: any }[] = [];
 
     for (const key of allowedKeys) {
-      const value = data[key];
-      // Ignore null, undefined, and empty strings
-      if (value !== undefined && value !== null && value !== '') {
-        activeFilters.push({ key, value });
+      if (key === 'token') {
```

Comment:
the room key generator was a pretty general-purpose class 🤔 but now you glued the token.EGLD logic inside of it. can you at least extract the custom logic to a new function ?